ocaml/xenopsd/xc/device.ml

```diff
@@ -1238,7 +1240,57 @@ module PCI = struct
                 shift_right_logical (add _page_size scan_size |> pred) 12
               )
             in
+            (* Linux disables the ROM BARs if not used and by default does
+               not set the start address.
+               Force it to set the address using "rom" file.
+               This method works also with lockdown mode enabled. *)
+            let enable_rom start =
+              (* read current configured ROM address to check if enabled *)
+              let config_fn = sysfs_pci_dev ^ "/config" in
+              let out = Bytes.make 4 '\x00' in
+              let fd =
+                Unix.openfile config_fn [Unix.O_RDONLY; Unix.O_CLOEXEC] 0
+              in
+              let current_addr =
+                finally
+                  (fun () ->
+                    Unix.(lseek fd 0x30 SEEK_SET) |> ignore ;
+                    (* this is a virtual file that control PCI configuration,
+                       it will either fail or return 4 *)
+                    Unix.(read fd out 0 4) |> ignore ;
+                    Nativeint.(
```

Comment:
Why do you prefer using nativeint over a known fixed size, while also using fixed-size constants?